python/mxnet/executor_manager.py

```diff
@@ -216,8 +225,23 @@ def __init__(self, sym, arg_names, param_names, ctx, slices, train_data, shared_
 
         self.train_execs = []
         for i in range(len(ctx)):
-            data_shapes = {k: tuple([slices[i].stop-slices[i].start] + list(v[1:]))
-                           for k, v in train_data.provide_data + train_data.provide_label}
+            data_shapes = {}
+            batch_size = 0
+            for k, v in train_data.provide_data:
+                if k == 'data':
```

Comment:
This is not good. 

In previous implements, provide 'data' in provide_data is not necessary. This change means we must have a 'data' element in provide_data. In one of my [task](http://www.jianshu.com/p/b11f3fa95bcb), I does not provide 'data'. 

I think there is other better ways to calculate batch_size.